chassembly/src/main/java/org/quiltmc/chasm/lang/internal/intrinsics/MapFunction.java

```diff
@@ -0,0 +1,45 @@
+package org.quiltmc.chasm.lang.internal.intrinsics;
+
+import java.util.Map;
+
+import org.quiltmc.chasm.lang.api.ast.ListNode;
+import org.quiltmc.chasm.lang.api.ast.MapNode;
+import org.quiltmc.chasm.lang.api.ast.Node;
+import org.quiltmc.chasm.lang.api.eval.Evaluator;
+import org.quiltmc.chasm.lang.api.eval.FunctionNode;
+import org.quiltmc.chasm.lang.api.exception.EvaluationException;
+
+/**
+ * The map intrinsic, which takes in a list and a function, and returns a new list of the result of applying the
+ * function to each element in the list.
+ */
+public class MapFunction extends IntrinsicFunction {
+    @Override
+    public Node apply(Evaluator evaluator, Node arg) {
+        if (!(arg instanceof MapNode)) {
+            throw createArgsException(arg);
+        }
+
+        Map<String, Node> args = ((MapNode) arg).getEntries();
+        Node list = args.get("list");
+        Node function = args.get("function");
```

Comment:
"function" is a bit of a long argument name, could it be shortened to just "func", or something along those lines?

Reply:
Agreed, `func` still gets it across in far less characters.